tern/formats/html/generator.py

```diff
@@ -0,0 +1,261 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: BSD-2-Clause
+
+"""
+HTML document generator
+"""
+
+import logging
+
+from tern.formats import generator
+from tern.report.content import get_tool_version, get_licenses_only
+from tern.utils import constants
+
+
+# global logger
+logger = logging.getLogger(constants.logger_name)
+
+
+css ='''
+<style>
+ul, #myUL {
+  list-style-type: none;
+}
+li {
+    padding-top: 5px;padding-bottom: 5px;
+}
+#myUL {
+  margin: 0;
+  padding: 0;
+  
+}
+.caret {
+  cursor: pointer;
+  user-select: none;
+  
+  font-family: 'Oswald', sans-serif;
+}
+.caret::before {
+  content: " \\25B6";
+  color: ;
+  display: inline-block;
+  margin-right: 6px;
+}
+.caret-down::before {
+  transform: rotate(90deg);
+}
+.nested {
+  display: none;
+}
+.active {
+  display: block;
+} 
+.header {
+  text-align: left;
+  background: #f1f1f1;
+}
+</style>\n
+'''
+
+
+js= ''' 
+<script>
+var toggler = document.getElementsByClassName("caret");
+var i;
+for (i = 0; i < toggler.length; i++) {
+  toggler[i].addEventListener("click", function() {
+    this.parentElement.querySelector(".nested").classList.toggle("active");
+    this.classList.toggle("caret-down");
+  });
+} 
+</script>\n
+'''
+
+
+head = """
+<!DOCTYPE html>
+<head>
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<link href="https://fonts.googleapis.com/css2?family=Inconsolata:wght@300&family=Oswald&display=swap" rel="stylesheet"><link href="https://fonts.googleapis.com/css2?family=Josefin+Sans:ital,wght@1,300&display=swap" rel="stylesheet"> 
+<link href="https://fonts.googleapis.com/css2?family=Inconsolata&display=swap" rel="stylesheet"> 
+%s \n
+</head>
+<body>
+<div class="header">
+<img src="https://raw.githubusercontent.com/tern-tools/tern/master/docs/img/tern_logo.png" height="60px">\n
+<br>
+</div>\n
+<div style="font-family: \'Inconsolata\', monospace;">
+<p>
+Tern at %s
+<p>
+The following report was generated for "%s" image.
+</div>
+"""
+
+
+def image_handler(listObj, indent):
+    html_string = ''
+    for i in listObj:
+        html_string = html_string + dict_handler(i,indent+1)
+    return html_string
+
+
+def manifest_handler(listObj, indent):
```

Comment:
Can you add function descriptions to these? They can be short but it helps with maintainability and helping newcomers to the project understand what is going on.